core/src/main/java/io/confluent/kafka/schemaregistry/rest/resources/ConfigResource.java

```diff
@@ -84,7 +84,11 @@ public ConfigUpdateRequest updateSubjectLevelConfig(
       throw new RestInvalidCompatibilityException();
     }
     try {
-      Map<String, String> headerProperties = requestHeaderBuilder.buildRequestHeaders(headers);
+      if (schemaRegistry.config() == null) {
```

Comment:
shouldn't this be asserted in the constructor?

Reply:
Isn't this an anti-pattern - to throw an exception from a constructor?